ui/StatusQ/src/movablemodel.cpp

```diff
@@ -199,6 +187,18 @@ bool MovableModel::synced() const
 
 void MovableModel::resetInternalData()
```

Comment:
Hey, I looked at the problem and came to a few conclusions.

The source of both problems (duplicated connections, connections inadequate to `m_synced` flag) is in `resetInternalData`.

In this method `m_synced` is set to true, but the actual sync is not done.
To fix it my proposition is as follows (referring to the original code as a baseline):

Instead of calling `resetInternalData()` in `syncOrder` it should call:
```
    m_indexes.clear();
    m_synced = true;
    emit syncedChanged();
```
Then the whole impl of `resetInternalData()` should look like that:
```
 QAbstractListModel::resetInternalData();
 m_indexes.clear();

 if (!m_synced)
   syncOrder(); 
```

Then there is no need to use `Qt::UniqueConnection` - it masks the real problem but it's not the solution of the root cause.

`connectSignalsForSyncedState` doesn't have to call `disconnect`. All disconnections are handled by `disconnect(m_sourceModel, nullptr, this, nullptr);`.

Also it would be nice to add test failing before introducing those changes.

Reply:
Thanks for having a look!

Good point! I'll look into this and add the tests.